torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.cpp

```diff
@@ -0,0 +1,253 @@
+#include <mutex>
+#include <regex>
+
+#include <torch/csrc/jit/codegen/cuda/instrumentation.h>
+#include "torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.h"
+#include "torch/csrc/jit/codegen/cuda/kernel_db/utils.h"
+#include "torch/csrc/jit/codegen/cuda/utils.h"
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+static std::mutex kernel_db_lock;
+
+KernelDb::KernelDb(bool _disabled)
+    : disabled_(_disabled),
+      initialized_(false),
+      kernel_map_(),
+      kernel_db_path_(),
+      kernel_db_txt_file_() {}
+
+bool KernelDb::open(
```

Comment:
I wonder if open should be in a big `try` `catch` block in case there's any completely unexpected errors when trying to read the file.

Reply:
Added.